### PR TITLE
343 todo list component needed

### PR DESCRIPTION
**Pull request by akinolu52**
A pr to solve this [issue](https://github.com/web3ui/web3uikit/issues/343)
Screenshot 2022-04-04 at 12 32 38

### CODE REVIEW

**Review by locothedev — CHANGES_REQUESTED** [review]

File: src/components/Todo/Todo.styles.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    display: flex;
+    gap: 16px;
+    margin-bottom: 20px;
+`;
+
+export const DivStyledContent = styled.section<TStyleProps>`
+    display: grid;
+    gap: 16px;
+    grid-template-columns: ${({ fullWidth }) => `repeat(${fullWidth ? 1 : 4}, minmax(0, 1fr))`};
+`;
```
[the remark this answers is not in the document]
Comment (akinolu52) [reply src/components/Todo/Todo.styles.ts:18]: okay

File: src/components/Todo/Todo.styles.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+export const SectionStyled = styled.section`
+    background-color: ${color.blueLight};
```
[the remark this answers is not in the document]
Comment (akinolu52) [reply src/components/Todo/Todo.styles.ts:21]: did you look at the figma design?

should background color come in as a props?

File: src/components/Todo/Todo.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        };
+        setLists((prevTodo) => [...prevTodo, todo]);
+        setInputValue('');
+    };
+
+    return (
+        <SectionStyled data-testid="todo-comp">
+            <DivStyled>
+                <Input
+                    label="Enter IP"
```
Comment (locothedev) [inline src/components/Todo/Todo.tsx:46]: make it modular

File: src/components/Todo/Todo.styles.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (locothedev) [reply src/components/Todo/Todo.styles.ts:21]: not as a prop I think its enough when user do this:

`<div style={{background-color: 'blue'}}>
  <Todo  {...todo} />
</div>`

File: src/components/Todo/Todo.tsx (hunk 3 above)
Comment (akinolu52) [reply src/components/Todo/Todo.tsx:46]: is there a need to accept button text ?

Comment (locothedev) [reply src/components/Todo/Todo.tsx:46]: not really but could be handy in future. Up to you my bro

Comment (akinolu52) [reply src/components/Todo/Todo.tsx]: okay

File: src/components/Todo/Todo.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                <Input
+                    label={label}
+                    size="large"
+                    validation={{
+                        regExp: pattern,
+                    }}
+                    onChange={handleInputChange}
+                />
+                <Button
+                    theme="primary"
```
[the remark this answers is not in the document]
Comment (akinolu52) [reply src/components/Todo/Todo.tsx]: nice one @BillyG83 , done

**Pull request merged**